python/hudi/_internal.pyi

```diff
@@ -349,3 +349,18 @@ def build_hudi_table(
         HudiTable: An instance of hudi table.
     """
     ...
+@dataclass(init=False)
+class HudiDataSource:
```

Comment:
would love to see this refactored into another `.pyi` for better code organization. it'll be optional here and you can do it separately if you like.

Reply:
I tried to do this but got some errors, so I think I will try to do it in a different PR.